### PR TITLE
feat: FlightSQL support extension types

### LINKED ISSUE #208: feat: support datatype variant
Variant field is json in REST API, but not in FlightSQL.

### PR DESCRIPTION
**Pull request by b41sh**
FlightSQL support `EmptyArray`, `EmptyMap`, `Bitmap` and `Variant`

Close #208

### CODE REVIEW

**Comment by everpcpc** [comment]
`test-bendsql` seems failed.

**Comment by b41sh** [comment]
> `test-bendsql` seems failed.

test has fixed, PTAL @everpcpc

File: cli/tests/00-base.result [hunk 1]
```diff
@@ -1,9 +1,10 @@
-b	2	false
-a	1	true
-0	99999	100000
+b	2	false	"{""k"":""v""}"
```
Comment (everpcpc) [inline cli/tests/00-base.result:1]: The result seems weird.

Comment (b41sh) [reply cli/tests/00-base.result:1]: this is because the `QuoteStyle` of tsv is `Necessary`, string will add extra comma.

**Pull request merged**